Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -961,6 +1065,16 @@ public extension ChatChannelController {
         }
     }
 
+    private func updateOldestFetchedMessageId(with payload: ChannelPayload) {
```

Comment:
They are actually not. It is called on the main thread and in the Updater thread. The likelihood of having concurrency issues, in this case, is very rare since the requests take some time always. The previous existing code was already working this way and without any concurrency issues. So for now I don't think we should worry about this and make the code unnecessarily more complex, but let me know what you think 👍

Reply:
I actually unit-tested this, and there is no concurrency issue